frontend/src/metabase/components/ItemsTable/Columns.tsx

```diff
@@ -134,29 +136,41 @@ export const Columns = {
       item: CollectionItem;
       testIdPrefix?: string;
       includeDescription?: boolean;
-      onClick?: () => void;
-    }) => (
-      <ItemNameCell data-testid={`${testIdPrefix}-name`}>
-        <ItemLink to={item.getUrl()} onClick={onClick}>
-          <EntityItem.Name name={item.name} variant="list" />
-          <PLUGIN_MODERATION.ModerationStatusIcon
-            size={16}
-            status={item.moderated_status}
-          />
-          {item.description && includeDescription && (
-            <DescriptionIcon
-              name="info"
+      onClick?: (item: CollectionItem, event: MouseEvent) => void;
+    }) => {
+      const ItemLinkComponent = onClick
```

Comment:
If there's a better way to do this I'm all ears

Reply:
I remember I had to do something similar for when I needed something that could either be a button or a link 🙃